### ISSUE TITLE
AttributeError: 'bool' object has no attribute 'get' when generating models from OpenAPI3 json schema

### ISSUE THREAD

**Issue by Darion** [koxudaxi/datamodel-code-generator#1536]
**Describe the bug**
AttributeError: 'bool' object has no attribute 'get' when generating models from OpenAPI3 json schema

**To Reproduce**

Schema: https://github.com/OAI/OpenAPI-Specification/blob/main/schemas/v3.1/schema.yaml

Used commandline:
```
$ git clone https://github.com/OAI/OpenAPI-Specification/
$ cd OpenAPI-Specification/schemas/v3.1
$ datamodel-codegen --input schema.yaml --output oas31models.py
```

Expected: Generated models, no errors

Got:

```
The input file type was determined to be: jsonschema
This can be specificied explicitly with the `--input-file-type` option.
/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/pydantic/main.py:309: UserWarning: Pydantic serializer warnings:
  Expected `Union[list[definition-ref], definition-ref, bool]` but got `JsonSchemaObject` - serialized value may not be as expected
  return self.__pydantic_serializer__.to_python(
Traceback (most recent call last):
  File "/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/datamodel_code_generator/__main__.py", line 766, in main
    generate(
  File "/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/datamodel_code_generator/__init__.py", line 433, in generate
    results = parser.parse()
              ^^^^^^^^^^^^^^
  File "/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/datamodel_code_generator/parser/base.py", line 1022, in parse
    self.parse_raw()
  File "/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1606, in parse_raw
    self._parse_file(self.raw_obj, obj_name, path_parts)
  File "/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1702, in _parse_file
    model_name, JsonSchemaObject.parse_obj(models), path
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/typing_extensions.py", line 2562, in wrapper
    return __arg(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/pydantic/main.py", line 976, in parse_obj
    return cls.model_validate(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/pydantic/main.py", line 504, in model_validate
    return cls.__pydantic_validator__.validate_python(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/datamodel_code_generator/parser/jsonschema.py", line 275, in __init__
    super().__init__(**data)
  File "/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/pydantic/main.py", line 165, in __init__
    __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)
  File "/home/alex/work/OpenAPI-Specification/schemas/v3.1/.venv/lib/python3.11/site-packages/datamodel_code_generator/parser/jsonschema.py", line 199, in validate_exclusive_maximum_and_exclusive_minimum
    exclusive_maximum: Union[float, bool, None] = values.get('exclusiveMaximum')
                                                  ^^^^^^^^^^
AttributeError: 'bool' object has no attribute 'get'
```

**Version:**
 - OS: Linux
 - Python version: 3.11.3
 - datamodel-code-generator version: 22.4 (tried master branch too)
 - pydantic version: 2.3.0
 
**Additional context**

This also fails:

```
$ datamodel-code-generator --url https://raw.githubusercontent.com/oasis-tcs/sarif-spec/main/Schemata/sarif-schema-2.1.0.json --output model.py 
```

Looks like  #1442, but it's closed and does not have pydantic warning.

**Comment by sbussard-vareto** [comment]
bump

**Comment by sbussard-vareto** [comment]
Same failure, but for json-schema

**Comment by sbussard-vareto** [comment]
I do not know the down stream effects of this change, but it seems to work as a workaround
https://github.com/koxudaxi/datamodel-code-generator/compare/main...sbussard-vareto:datamodel-code-generator:patch-1

**Comment by dnfsv** [comment]
Also seeing same failure for json-schema

**Comment by mmwinther** [comment]
I experience this same error, when running with Python 3.10 or 3.11. The official Docker container works fine and that uses Python 3.12. Could 3.12 compatibility have broken older versions?

**Comment by tractorcow** [comment]
hi @sbussard-vareto thanks your fix helped me out. Is it possible to get this turned into a pull request?

**Comment by sbussard-vareto** [comment]
@koxudaxi already took care of it!
https://github.com/koxudaxi/datamodel-code-generator/pull/1691

**Comment by koxudaxi** [comment]
@Darion  @dnfsv  @mmwinther  @tractorcow  @sbussard-vareto 
I have released a fixed version  [0.24.0](https://github.com/koxudaxi/datamodel-code-generator/releases/tag/0.24.0)
Thank you so much!!
